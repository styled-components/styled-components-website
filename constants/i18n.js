const DEFAULT_TRANSLATION = 'translations'
const NAV_TRANSLATION = 'nav'
const HOME_TRANSLATION = 'home'
const DOCS_TRANSLATION = 'docs'
const DOCS_BASICS_TRANSLATION = 'docsBasics'
const DOCS_ADVANCED_TRANSLATION = 'docsAdvanced'
const DOCS_API_TRANSLATION = 'docsApi'
const DOCS_FAQS_TRANSLATION = 'docsFaqs'

const TRANSLATIONS = [
  DEFAULT_TRANSLATION,
  NAV_TRANSLATION,
  HOME_TRANSLATION,
  DOCS_TRANSLATION,
  DOCS_BASICS_TRANSLATION,
  DOCS_ADVANCED_TRANSLATION,
  DOCS_API_TRANSLATION,
  DOCS_FAQS_TRANSLATION,
]

const DEFAULT_LANGUAGE = 'en'

const LANGUAGES = [
  DEFAULT_LANGUAGE,
  'es',
]

module.exports = {
  DEFAULT_TRANSLATION,
  NAV_TRANSLATION,
  TRANSLATIONS,
  HOME_TRANSLATION,
  DOCS_TRANSLATION,
  DOCS_BASICS_TRANSLATION,
  DOCS_ADVANCED_TRANSLATION,
  DOCS_API_TRANSLATION,
  DOCS_FAQS_TRANSLATION,
  LANGUAGES,
  DEFAULT_LANGUAGE,
}
