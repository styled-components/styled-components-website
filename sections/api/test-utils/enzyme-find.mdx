### `enzymeFind` | v4

A convenience method for finding instances of a particular styled component within an enzyme wrapper.

```jsx
import { mount } from 'enzyme'
import styled from 'styled-components'
import { enzymeFind } from 'styled-components/test-utils'

const Foo = styled.div`
  color: red;
`

const wrapper = mount(
  <div>
    <Foo>bar</Foo>
  </div>
)

enzymeFind(wrapper, Foo)
```
