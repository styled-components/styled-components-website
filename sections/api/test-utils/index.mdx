import Find from './find.mdx'
import FindAll from './find-all.mdx'
import EnzymeFind from './enzyme-find.mdx'

## Test Utilities

<Find />

<FindAll />

<EnzymeFind />
