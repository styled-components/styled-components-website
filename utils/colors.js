import lighten from 'polished/lib/color/lighten';
import darken from 'polished/lib/color/darken';
import shade from 'polished/lib/color/shade';

export const paleGrey = '#efefef';
export const lightGrey = 'rgba(20, 20, 20, 0.1)';
export const darkGrey = darken(0.05, '#282a36');
export const grey = '#282a36';

export const red = '#ff5555';
export const violetRed = 'rgb(219, 112, 147)';
export const darkVioletRed = darken(0.1, 'rgb(219, 112, 147)');
export const lightVioletRed = lighten(0.31, 'rgb(219, 112, 147)');

export const gold = shade(0.9, 'rgb(243, 182, 97)');
