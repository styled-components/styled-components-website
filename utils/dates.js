export const getFormattedDate = (unformattedDate) => {
  const date = new Date(unformattedDate);
  const year = date.getFullYear();
  let month = (1 + date.getMonth()).toString();
  month = month.length > 1 ? month : '0' + month;
  let day = date.getDate().toString();
  day = day.length > 1 ? day : '0' + day;

  return day + '/' + month + '/' + year;
};
