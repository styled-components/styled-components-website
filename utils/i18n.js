import i18n from 'i18next'
import LanguageDetector from 'i18next-browser-languagedetector'
import { reactI18nextModule } from 'react-i18next'

import translations from '../locales'

i18n
  .use(LanguageDetector)
  .use(reactI18nextModule)
  .init({
    fallbackLng: 'en',

    // have a common namespace used around the full app
    ns: ['translations', 'homepage'],
    defaultNS: 'translations',

    debug: process.env.NODE_ENV !== 'production',

    interpolation: {
      escapeValue: false, // not needed for react!!
    },

    react: {
      wait: true
    },

    resources: { ...translations }
  })

export default i18n
