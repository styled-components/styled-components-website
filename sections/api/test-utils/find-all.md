### `findAll` | v3

A convenience method to find all instances of a styled component's rendered DOM node within a given DOM root.

```js
import styled from 'styled-components'
import { findAll } from 'styled-components/test-utils'

const Foo = styled.div`
  color: ${props => props.color};
`

/**
 * Somewhere in your app:
 *
 * ReactDOM.render(
 *   <main>
 *     <Foo color="red" />
 *     <Foo color="green" />
 *   </main>, document.body
 * );
 */

// retrieves a NodeList of instances of "Foo" in the body (querySelectorAll under the hood)
findAll(document.body, Foo) // NodeList<HTMLDivElement> | null
```
