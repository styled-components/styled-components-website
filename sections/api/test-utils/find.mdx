### `find` | v3

A convenience method to find a single instance of a styled component's rendered DOM node within a given DOM root.

```js
import styled from 'styled-components'
import { find } from 'styled-components/test-utils'

const Foo = styled.div`
  color: red;
`

/**
 * Somewhere in your app:
 *
 * ReactDOM.render(
 *   <main>
 *     <Foo />
 *   </main>, document.body
 * );
 */

// retrieves the first instance of "Foo" in the body (querySelector under the hood)
find(document.body, Foo) // HTMLDivElement | null
```
