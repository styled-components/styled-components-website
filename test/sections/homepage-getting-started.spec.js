import React from 'react'
import { mount } from 'enzyme'
import 'jest-styled-components'
import { I18nextProvider } from 'react-i18next'
import { mockRouter } from '../../utils/test-utils'
import HomepageGettingStarted from '../../sections/homepage-getting-started'
import i18n from '../helpers/i18n'

mockRouter()

const wrapper = (props, language) => mount(
  <I18nextProvider i18n={i18n} initialLanguage={language}>
    <HomepageGettingStarted {...props} />
  </I18nextProvider>
)

test('HomepageGettingStarted renders correctly', () => {
  expect(wrapper({}, 'en').find(HomepageGettingStarted)).toMatchSnapshot()
})

test('HomepageGettingStarted renders in spanish', () => {
  expect(wrapper({}, 'es').find(HomepageGettingStarted)).toMatchSnapshot()
})
