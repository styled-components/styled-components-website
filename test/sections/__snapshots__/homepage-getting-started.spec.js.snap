// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`HomepageGettingStarted renders correctly 1`] = `
.c0 {
  width: 56.888888888888886rem;
  max-width: 100%;
  margin: 0 auto;
  padding: 5rem 2.2222222222222223rem 1.6666666666666667rem 2.2222222222222223rem;
  box-sizing: border-box;
  font-family: -apple-system,BlinkMacSystemFont,"Segoe UI",Helvetica,Arial,sans-serif,"Apple Color Emoji","Segoe UI Emoji","Segoe UI Symbol";
  -webkit-transition: -webkit-transform 150ms ease-out;
  -webkit-transition: transform 150ms ease-out;
  transition: transform 150ms ease-out;
}

.c1 {
  text-align: center;
}

.c2 {
  margin: 0 0.5em 3em;
  height: 1.5em;
}

.c3 {
  border-radius: 3px;
  padding: 0.25em 1em;
  margin: 0 1em;
  background: transparent;
  color: palevioletred;
  border: 2px solid palevioletred;
}

.c4 {
  border-radius: 3px;
  padding: 0.25em 1em;
  margin: 0 1em;
  background: transparent;
  color: palevioletred;
  border: 2px solid palevioletred;
  background: palevioletred;
  color: white;
}

.c5 {
  border-radius: 3px;
  padding: 0.25em 1em;
  background: transparent;
  color: palevioletred;
  border: 2px solid palevioletred;
}

@media (max-width:62.5em) {
  .c0 {
    padding: 3.888888888888889rem 1.1111111111111112rem 1.6666666666666667rem 1.1111111111111112rem;
  }
}

<HomepageGettingStarted>
  <I18n
    ns="home"
  >
    <styled.div>
      <div
        className="c0"
      >
        <styled.div>
          <div
            className="c1"
          >
            <a
              href="https://github.com/styled-components/styled-components"
            >
              <styled.img
                alt="Stars on GitHub"
                src="https://img.shields.io/github/stars/styled-components/styled-components.svg?style=social&label=Star&maxAge=3600"
              >
                <img
                  alt="Stars on GitHub"
                  className="c2"
                  src="https://img.shields.io/github/stars/styled-components/styled-components.svg?style=social&label=Star&maxAge=3600"
                />
              </styled.img>
            </a>
            <a
              href="https://www.npmjs.com/package/styled-components"
            >
              <styled.img
                alt="Current Version"
                src="https://img.shields.io/npm/v/styled-components.svg"
              >
                <img
                  alt="Current Version"
                  className="c2"
                  src="https://img.shields.io/npm/v/styled-components.svg"
                />
              </styled.img>
            </a>
            <styled.img
              alt="Monthly downloads"
              src="https://img.shields.io/npm/dm/styled-components.svg?maxAge=3600"
            >
              <img
                alt="Monthly downloads"
                className="c2"
                src="https://img.shields.io/npm/dm/styled-components.svg?maxAge=3600"
              />
            </styled.img>
            <styled.img
              alt="Gzipped size"
              src="http://img.badgesize.io/https://unpkg.com/styled-components/dist/styled-components.min.js?compression=gzip&label=gzip%20size"
            >
              <img
                alt="Gzipped size"
                className="c2"
                src="http://img.badgesize.io/https://unpkg.com/styled-components/dist/styled-components.min.js?compression=gzip&label=gzip%20size"
              />
            </styled.img>
            <a
              href="https://spectrum.chat/styled-components"
            >
              <styled.img
                alt="Join the community on Spectrum"
                src="https://withspectrum.github.io/badge/badge.svg"
              >
                <img
                  alt="Join the community on Spectrum"
                  className="c2"
                  src="https://withspectrum.github.io/badge/badge.svg"
                />
              </styled.img>
            </a>
          </div>
        </styled.div>
        <styled.div>
          <div
            className="c1"
          >
            <styled.button
              onClick={[Function]}
            >
              <button
                className="c3"
                onClick={[Function]}
              >
                Normal button
              </button>
            </styled.button>
            <styled.button
              onClick={[Function]}
              primary={true}
            >
              <button
                className="c4"
                onClick={[Function]}
              >
                Primary button
              </button>
            </styled.button>
          </div>
        </styled.div>
        <styled.div>
          <div
            className="c1"
          >
            <button>
              I'm a &lt;Button /&gt;!
            </button>
          </div>
        </styled.div>
        <styled.div>
          <div
            className="c1"
          >
            <styled.button>
              <button
                className="c5"
              >
                I'm a styled &lt;Button /&gt;
              </button>
            </styled.button>
          </div>
        </styled.div>
      </div>
    </styled.div>
  </I18n>
</HomepageGettingStarted>
`;
